### ISSUE TITLE
Fix inconsistent and unclear order of JSX attributes

### ISSUE THREAD

**Issue by Ishirak** [versionpress/versionpress#1086]
This issue is part of a frontend code review, see #1083.

The front end codebase contains inconsistent and unclear order of JSX attributes.

Multiple JSX tags and attributes are in some components on the same line, in other components, or even in other parts of the same component, they aren't.

For example,
```JSX
<a className='BulkActionPanel-clear' href='#' onClick={this.onClearSelection.bind(this)}>clear selection</a>
```
can be converted into
```JSX
<a 
    className="bulk-action-panel-clear"
    href="#" 
    onClick={this.onClearSelection}
>
    clear selection
</a>
```
which is much cleaner and readable.

**Comment by vasek17** [comment]
In this case I would prefer (e.g. [here](https://github.com/versionpress/versionpress/blob/8778ceb405fc69cf044baf66d08fa76981c363cb/frontend/src/CommitPanel/CommitPanelCommit.react.tsx#L49))
```javascript
<a 
    className="bulk-action-panel-clear"
    href="#" 
    onClick={this.onClearSelection}
>clear selection</a>
``` 
or
```javascript
<a className="bulk-action-panel-clear" href="#" onClick={this.onClearSelection}>
    clear selection
</a>
``` 
when the text would be longer or it would be block.

I try to write the component on a single line if it's readable otherwise display it multi-line and indent.
This example is an edge case - it is ok for me both single line and multi-line.

**Comment by borekb** [comment]
This is a small stylistic issue where different people will probably have different tastes (I, for example, prefer Vasek's last example). I don't think we need a hard rule here, just do whatever seems sensible and I, personally, don't have a problem reading a code where the styles are slightly mixed and matched as long as the code is readable (and Vasek's last example is perfectly fine by me).

**Comment by vasek17** [comment]
@Ishirak: I don't think I get your point so maybe one more example.
If the tag has a lot of attributes it should be multi-line. If it contains block or the text is longer is could look like this:
```javascript
<a 
    className="bulk-action-panel-clear"
    href="#" 
    onClick={this.onClearSelection}
    anotherAttribute="looooooong one"
    yetAnother=true
>
    Veeeeery long lorem ipsum dolor sit amet text. ....
</a>
```

However it does not make sense to me to format small tags (extreme case) like this:
```javascript
<a 
    href="#"
>
    close
</a>
```

In this case 
```javascript 
<a href="#">close</a>
```
would be preferred.

I would agree not to have a hard rule as long as the code is readable. 
Any rule in this case could lead to code which is difficult to read (e.g. unnecessarily long horizontally or vertically).

**Comment by Ishirak** [comment]
@vasek17 your first example is a great one, clear and readable.

Your second example is a little bit too much, you're right, even I do not use it like that.
Personally I use something like this:
```JSX
<a href="#">
    close
</a>
```
I've developed my JSX style by myself for the past year, but [AirBnb's](https://github.com/airbnb/javascript/tree/master/react#alignment) is in this example exactly what I use.

**Comment by vasek17** [comment]
I don't think Airbnb style guide is any strict in this case.
E.g.
```javascript
render() {
    const body = <div>hello</div>;
    return <MyComponent>{body}</MyComponent>;
}
```
(https://github.com/airbnb/javascript/tree/master/react#parentheses) 

or
```javascript
render() {
    return <a href={this.props.url} data-id={this.props.id}>{this.props.text}</a>
}
```
(https://github.com/airbnb/javascript/tree/master/react#ordering)

**Comment by Ishirak** [comment]
@vasek17 Yes, I know, because of AirBnb's inconsistency in the guide I didn't include it in the main post, I only wanted to let you know about the alignment part which I use.
I think the best approach would be according to your post, especially this part:
> I would agree not to have a hard rule as long as the code is readable. 
> Any rule in this case could lead to code which is difficult to read (e.g. unnecessarily long horizontally or vertically).

**Comment by vasek17** [comment]
👍 
Feel free to reformat any line that is difficult to read (e.g. the one proposed in the first post).

**Comment by borekb** [comment]
Good conclusion. The reformatting doesn't need to be done as a dedicated pull request (just do it continuously as you encounter hard to read code) so I'm closing this.

**Comment by borekb** [comment]
:ship: Shipped in [4.0-alpha1](https://github.com/versionpress/versionpress/releases/tag/4.0-alpha1).
